### DISCUSSION TITLE
Pivot functionality

### DISCUSSION THREAD

**Discussion by cmrts** [hosseinmoein/DataFrame#215] [Ideas]
Hello. I was wondering if you have plans to create pivot table functionality?

That would be the equivalent of df.pivot()

regards

**Comment by hosseinmoein** [comment]
It is one of the items on my lit. But implementing a pivot table in C++ is not that simple, since C++ is statically typed.
I have to find time to think about a solution/design and implement it. I am very short in time.

**Reply by cmrts** [reply]
Thanks. I was considering doing a similar functionality in C#, and concluded that I needed three queries:
1) one group-by, by both rows/columns together => giving you all valid combination cells and their measures. The problem is then presentational transformation 
2) one group by rows (ignore measures)=> isolate unique row labels
3) one group by columns (ignore measures)=> isolate unique column labels

Then it is a matter of combining all above in a 2D table.  In C# you can resize a 2-D array using the dimensions of 2/3 labels and populate with content from 1.

I didn’t have the group by, which was the difficult bit, and you are ahead of where I was.

I was considering C# as I haven’t got experience with C++, so may sound easier with my limited knowledge than it actually is in C++, I appreciate.

**Reply by hosseinmoein** [reply]
Yeah I have the group by stuff and all. The logic of pivot table is straightforward but rendering it in c++ is tricky because c++ containers can have only one type — in other words c++ is statically typed
